lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -1324,7 +1324,7 @@ TNonblockingIOThread::~TNonblockingIOThread() {
     ownEventBase_ = false;
   }
 
-  if (listenSocket_ >= 0) {
+  if (listenSocket_ >= 0 && listenSocket_ != THRIFT_INVALID_SOCKET) {
```

Comment:
Perhaps the only test needed is to see if the socket is not an invalid socket?  Is the >= 0 test necessary at this point?